tests/acceptance/async_test_base.py

```diff
@@ -91,11 +92,10 @@ def on_open_error(self, connection, error):
 
     def on_timeout(self):
         """called when stuck waiting for connection to close"""
-        self.logger.info('on_timeout at %s', datetime.utcnow())
-        # force the ioloop to stop
-        self.logger.debug('on_timeout called')
-        self.connection.ioloop.stop()
-        raise AssertionError('Test timed out')
+        self.logger.info('on_timeout called at %s', datetime.utcnow())
+        # Initiate cleanup
+        self.timeout = None  # the dispatcher should have removed it
+        self.stop()
```

Comment:
heuh yes.. I had done this #678 totally based on your input if I remember correctly ;)